lib/components/map/connected-stop-viewer-overlay.js

```diff
@@ -21,6 +25,16 @@ class StopViewerOverlay extends MapLayer {
 
   updateLeafletElement () {}
 
+  componentDidUpdate (prevProps) {
+    const {leaflet, stopData} = this.props
+    // If a new stop is clicked, close the stop viewer popup
+    if (stopData?.id !== prevProps?.stopData?.id) {
+      leaflet.map && leaflet.map.eachLayer((layer) => {
+        if (layer.options.pane === 'popupPane') layer._closeButton.click()
```

Comment:
There are two existing ways to dismiss a popup, could you try either:
- by calling `setMapPopupLocation({ location: null })` (as in the `DefaultMap` component),
- by calling `leaflet.map.closePopup()` (as in the `SetFromToButtons` component).

Reply:
Very good call thank you! Much cleaner now in 718b9acd450d65c9fd1552edd7b464a6c021d387